internal/worker/task_handler.go

```diff
@@ -152,6 +154,51 @@ func (t *TaskHandler) setNextBootDevice(ctx context.Context, bmc device.Queryor)
 	return t.successful(ctx, "next boot device set successfully: "+t.task.Parameters.ActionParameter)
 }
 
+func (t *TaskHandler) validateFirmware(ctx context.Context, bmc device.Queryor) error {
+	t.loggerEntry.Info("starting firmware validation")
+
+	deadline := time.Now().Add(time.Duration(t.task.Parameters.ValidateFirmwareTimeout * uint(time.Second)))
```

Comment:
Consider making the `rivets` parameter a `time.Duration` there? Then you can avoid this (and blame the reviewer of that PR, he should have caught it 😅 ).

Reply:
I had actually thought about that but was under the impression those parameters were getting passed around in JSON blobs and thus might need to use primitive types instead of more structured/native ones...I gather that's not actually a concern?

Reply:
`time.Duration` serializes down to `uint64` (I think, it's nanoseconds) on the wire, but the ser/des routines for `time.Duration` mask that from you. It would be more of a concern if we had non-Golang clients in the mix.